Shared Cache Plugin won't configure locally

```
binaryninja-api/view/sharedcache [ dev][△ v3.31.3]
❯ BN_API_PATH=/Users/visual/Developer/binaryninja-api cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -S . -B build
-- The C compiler identification is AppleClang 16.0.0.16000026
-- The CXX compiler identification is AppleClang 16.0.0.16000026
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMAKE_PREFIX_PATH is: /Users/visual/Qt/6.7.2/clang_64/lib/cmake
CMake Error at core/CMakeLists.txt:24 (get_target_property):
  get_target_property() called with non-existent target "binaryninjaapi".
Call Stack (most recent call first):
  core/CMakeLists.txt:46 (get_recursive_include_dirs)


CMake Error at api/CMakeLists.txt:42 (get_target_property):
  get_target_property() called with non-existent target "binaryninjaapi".
Call Stack (most recent call first):
  api/CMakeLists.txt:64 (get_recursive_include_dirs)


CMake Deprecation Warning at api/python/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- RCD:
CMake Error at workflow/CMakeLists.txt:41 (get_target_property):
  get_target_property() called with non-existent target "binaryninjaapi".
Call Stack (most recent call first):
  workflow/CMakeLists.txt:65 (get_recursive_include_dirs)


CMake Deprecation Warning at ui/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found OpenGL: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/System/Library/Frameworks/OpenGL.framework
-- Found WrapOpenGL: TRUE
-- Could NOT find WrapVulkanHeaders (missing: Vulkan_INCLUDE_DIR)

▓█████▄    ██████   ▄████▄     Shared Cache Plugin
▒██▀ ██▌ ▒██    ▒  ▒██▀ ▀█
░██   █▌ ░ ▓██▄    ▒▓█    ▄    Metadata Version:      3
░▓█▄  █▌   ▒   ██▒ ▒▓▓▄ ▄██▒   CMake Prefix Path:     /Users/visual/Qt/6.7.2/clang_64/lib/cmake
░▒████▓  ▒██████▒▒▒  ▓███▀ ░   Qt Version:            6.7.2
 ▒▒▓  ▒ ▒ ▒▓▒ ▒ ░░ ░▒ ▒  ░     Crash on Failure:      OFF
 ░ ▒  ▒ ░ ░▒  ░ ░  ░  ▒        Slideinfo Debug Tags:  OFF
 ░ ░  ░ ░  ░  ░  ░             REFCOUNT_DEBUG:

-- Configuring incomplete, errors occurred!
```

--- a/view/sharedcache/CMakeLists.txt
+++ b/view/sharedcache/CMakeLists.txt
@@ -2,11 +2,14 @@ cmake_minimum_required(VERSION 3.13 FATAL_ERROR)
 
 project(sharedcache)
 
-if((NOT BN_API_PATH) AND (NOT BN_INTERNAL_BUILD))
-    set(BN_API_PATH $ENV{BN_API_PATH} CACHE STRING "Path to Binary Ninja API source")
-    if(NOT BN_API_PATH)
-        message(FATAL_ERROR "Provide path to Binary Ninja API source in BN_API_PATH")
-    endif()
+if(NOT BN_INTERNAL_BUILD)
+    find_path(
+        BN_API_PATH
+        NAMES binaryninjaapi.h
+        HINTS ../.. binaryninjaapi $ENV{BN_API_PATH}
+        REQUIRED
+    )
+    add_subdirectory(${BN_API_PATH} binaryninjaapi)
 endif()
 
 if (NOT BN_INTERNAL_BUILD)

